OsmIntegrator/Services/OsmUpdater.cs

```diff
@@ -344,5 +369,40 @@ private void UpdateStopProperties(DbStop stop, List<Tag> newTags, ReportStop rep
         stop.Number = localRefTag.Value;
       }
     }
+
+    public async Task UpdateTileReferences(List<DbTile> tiles, ApplicationDbContext dbContext)
+    {
+      using (IDbContextTransaction transaction = await dbContext.Database.BeginTransactionAsync())
+      {
+
+        try
+        {
+          foreach (DbTile tile in tiles)
+          {
+            foreach (DbStop stop in tile.Stops)
+            {
+              double stopLat = stop.Lat;
+              double stopLon = stop.Lon;
+
+              Point<long> tileCoordinates = TilesHelper.WorldToTilePos(stopLon, stopLat, _zoomLevel);
+
+              if (tile.X != tileCoordinates.X && tile.Y != tileCoordinates.Y)
+              {
+                stop.Tile = dbContext.Tiles.FirstOrDefault(t => t.X == tileCoordinates.X && t.Y == tileCoordinates.Y);
```

Comment:
Tutaj mi chodziło o przypadek gdy nie ma kafelka z podanymi koordynatami, to należy go dodać np.
```
DbTile tile = dbContext.Tiles.FirstOrDefault(t => t.X == tileCoordinates.X && t.Y == tileCoordinates.Y)
if(tile != null)
{ stop.Tile = tile; }
else {
DbTile newTile = ...
stop.Tile = newTile;
}

Reply:
Trzeba dodać nowy kafelek najpierw na bazie, żeby dostać jego Id.

Reply:
Dodałem taki przypadek